File: LeetcodeProblems/blind-75/longest-substring-without-repeating-characters/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    const n = s.length;
    let maxLength = 0;
    let left = 0;
    const charSet = new Set();
    
    for (let right = 0; right < n; right++) {
        const currentChar = s[right];
        
        while (charSet.has(currentChar)) {
            charSet.delete(s[left]);
            left++;
        }
        
        charSet.add(currentChar);
        maxLength = Math.max(maxLength, right - left + 1);
    }
    
    return maxLength;
};

File: LeetcodeProblems/blind-75/longest-substring-without-repeating-characters/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        n = len(s)
        max_length = 0
        left = 0
        char_set = set()
        
        for right in range(n):
            current_char = s[right]
            
            while current_char in char_set:
                char_set.remove(s[left])
                left += 1
            
            char_set.add(current_char)
            max_length = max(max_length, right - left + 1)
        
        return max_length